src/query/functions-v2/src/scalars/comparison.rs

```diff
@@ -236,19 +236,37 @@ fn register_like(registry: &mut FunctionRegistry) {
         FunctionProperty::default(),
         |_, _| None,
         vectorize_regexp(|str, pat, map, _| {
+            let mut sub_strings: Vec<&str> = vec![];
             let pattern = if let Some(pattern) = map.get(pat) {
                 pattern
             } else {
                 let pattern_str = simdutf8::basic::from_utf8(pat).map_err(|err| {
                     format!("unable to convert the LIKE pattern to string: {err}")
                 })?;
+                sub_strings = pattern_str
+                    .split(|c: char| c == '%' || c == '_' || c == '\\')
+                    .collect();
+                sub_strings.retain(|&substring| !substring.is_empty());
                 let re_pattern = like_pattern_to_regex(pattern_str);
                 let re = Regex::new(&re_pattern)
```

Comment:
regex's time complex is `O(n)`; every char in the str will only be checked once. 

The real problem here is that we need to move `Regex::new(&re_pattern)` out of the loop.

Reply:
Oh, we already cached the pattern. Please ignore.

Reply:
but  the  regexp is cached, the sub_strings will be always empty in the iteration.